### ISSUE TITLE
error TS2339: Property 'startsWith' does not exist on type 'string'

### ISSUE THREAD

**Issue by rajinder-yadav** [Microsoft/TypeScript#12383]
tsc 2.0.10

**Code**

let tok = ":name";
if(tok.startsWith(":"))  <--- false error: error TS2339: Property 'startsWith' does not exist on type 'string'

```ts
// A *self-contained* demonstration of the problem follows...

```

**Expected behavior:**
should know this is a method on a JS string.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

**Actual behavior:**

**Comment by j-oliveras** [comment]
The method endsWith is declared into es6 (es2015). Make sure your target is es6 or higher (if target is not defined defaults to es3) or use lib option to include "es2015.core".

**Comment by rajinder-yadav** [comment]
Thanks, closing!

**Comment by marcus-petty** [comment]
We have this but still get the same error

```js
{
  "compilerOptions": {
    "target": "es2015",
    "module": "commonjs",
    "sourceMap": true
  },
  "include": [
  ],
  "exclude": [],
  "lib":[
    "es2015"
  ]
}
``` error TS2339: Property 'startsWith' does not exist on type 'string'.

**Comment by mhegazy** [comment]
`"lib"` should be part of `"compilerOptions"` and not a sibling . see http://www.typescriptlang.org/docs/handbook/compiler-options.html and http://json.schemastore.org/tsconfig

**Comment by marcus-petty** [comment]
thx
